### PR TITLE
Handling 403 forbidden (bug 28489)

### PR DESCRIPTION
**Pull request by giancorzo**
I added a case to handle forbidden users

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx [hunk 1]
```diff
@@ -79,10 +78,14 @@ export function withCurrentUser() {
         this.fetchCurrentUser();
       }
 
+      isUserLocked = (response) => {
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: 403 won't always mean the user is locked. so I don't think it makes sense to always extract this to a method

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx [hunk 2]
```diff
@@ -107,6 +110,12 @@ export function withCurrentUser() {
           // https://github.com/json-api-dotnet/JsonApiDotNetCore/issues/39
           // is resolved
           const response = await authenticatedGet('/api/users/current-user?include=organization-memberships,group-memberships');
+
+          if (this.isUserLocked(response)) {
+            console.debug('Current user is Locked');
+            throw new Error('Your user has been locked, contact you organization administrator');
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: why are you throwing an error?

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx [hunk 3]
```diff
@@ -128,12 +137,16 @@ export function withCurrentUser() {
         const { error, currentUser, isLoading } = this.state;
 
         if (error) {
+          debugger;
           return <PageError error={error} />;
         }
 
         if (isLoading) {
+
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: why are you adding so many blank lines everywhere?

File: source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { getToken } from './auth0';
 import { getCurrentOrganizationId } from './current-organization';
 
 export const tryParseJson = async (response) => {
+
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: ??? also unneeded change

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx (hunk 1 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: So if 403 is not the correct way to check if a user is Blocked should I try to read the error message and see if has "some keyword about blocked" ?

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx (hunk 2 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: I'm throwing an error so I can reuse the error page

        if (error) {
          return <PageError error={error} />;
        }

File: source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts (hunk 4 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: It's easier to read if we always add 1 line at the beginning of the function

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: I disagree, if there were a rule for this, I'd add it. :-p
But in general, try not to make changes in your PRs that don't directly affect your PRs.

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx (hunk 1 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: checking for 403 is correct, but 403 will not always mean the user is locked

403 means forbidden

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx (hunk 2 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: :+1:

File: source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts (hunk 4 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: ok will change it, I used to follow some rules from HAPI community

File: source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts [hunk 5]
```diff
@@ -2,8 +2,8 @@ import { getToken } from './auth0';
 import { getCurrentOrganizationId } from './current-organization';
 
 export const tryParseJson = async (response) => {
-  const text = await response.text();
 
+  const text = await response.text();
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: what's with this change?

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: I can't tell what's different? lol

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: oh, it changed lines? weird

Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/lib/fetch.ts]: I didn't change this one... just added some debugger and space

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx [hunk 6]
```diff
@@ -107,6 +106,12 @@ export function withCurrentUser() {
           // https://github.com/json-api-dotnet/JsonApiDotNetCore/issues/39
           // is resolved
           const response = await authenticatedGet('/api/users/current-user?include=organization-memberships,group-memberships');
+
+          if (response.status === 403) {
+            console.debug('Current user is Forbidden');
+            throw new Error('An error occurred: Forbidden. Please contact your organization administrator.');
```
[the remark this answers is not in the document]
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user.tsx]: Translation added

**Pull request merged**